File: contest/LeetCode_Weekly_Contest_201/Make_The_String_Great.java
class Solution {
    public String makeGood(String s) {
        char[] ret = new char[s.length()+10];
        int idx = 0;

        for (int i = 0; i < s.length(); ++i) {
            if (idx == 0) {
                ret[idx++] = s.charAt(i);
                continue;
            }
            char a = ret[idx-1];
            char b = s.charAt(i);

            if (('a' <= a && a <= 'z' && 'A' <= b && b <= 'Z' && a-'a' == b-'A') ||
                ('A' <= a && a <= 'Z' && 'a' <= b && b <= 'z' && a-'A' == b-'a'))
                idx--;
            else ret[idx++] = b;

        }
        return String.copyValueOf(ret, 0, idx);
    }
}

File: contest/LeetCode_Weekly_Contest_201/Make_The_String_Great.cpp
class Solution {
public:
    string makeGood(string s) {
        string ret;
        for (char c : s) {
            if (!ret.empty() && (
                ('a' <= ret.back() && ret.back() <= 'z' && 'A' <= c && c <= 'Z' && ret.back()-'a' == c-'A') || 
                ('A' <= ret.back() && ret.back() <= 'Z' && 'a' <= c && c <= 'z' && ret.back()-'A' == c-'a'))) 
                ret.pop_back();
            else ret += c;
        }
        return ret;
    }
};
